src/vsphere_cpi/lib/cloud/vsphere/vm_creator.rb

```diff
@@ -18,154 +18,170 @@ def initialize(client:, cloud_searcher:, cpi:, datacenter:, agent_env:, ip_confl
     end
 
     def create(vm_config)
-      cluster = vm_config.cluster
-      storage = StoragePicker.choose_ephemeral_storage(vm_config.ephemeral_datastore_name, vm_config.cluster.accessible_datastores, vm_config.vm_type)
+      created_vm = nil
+      vm_config.cluster_placements.each do |cluster_placement|
+        # Get the  cluster from the placement
+        cluster = cluster_placement.cluster
```

Comment:
change the name to vm placement

Reply:
This also I can keep for future changes. As it will lead to lot of changes in unit test. I want things to be just working unless it is a breaking bug